packages/stellar-axelar-std-derive/src/storage.rs

```diff
@@ -0,0 +1,176 @@
+
+    for attr in attrs {
+        let path_str = attr.path().to_token_stream().to_string();
+
+        match path_str.as_str() {
+            "instance" => {
+                storage_attributes.storage_type = StorageType::Instance;
+            }
+            "persistent" => {
+                storage_attributes.storage_type = StorageType::Persistent;
+            }
+            "temporary" => {
+                storage_attributes.storage_type = StorageType::Temporary;
+            }
+            "value" => {
+                if let Meta::List(list) = &attr.meta {
+                    if let Ok(ty) = list.parse_args::<Type>() {
+                        storage_attributes.value_type = Some(ty);
+                    }
+                }
+            }
+            _ => {}
+        }
+    }
+
+    storage_attributes
+}
+
+fn storage_fns(
+    enum_name: &Ident,
+    variant: &Variant,
+    storage_attrs: &StorageAttributes,
+) -> TokenStream {
+    let variant_ident = &variant.ident;
+
+    let (field_names, field_types) = match &variant.fields {
+        Fields::Unit => (vec![], vec![]),
+        Fields::Named(fields) => {
+            let names = fields.named.iter().map(|f| &f.ident).collect();
+            let types = fields.named.iter().map(|f| &f.ty).collect();
+            (names, types)
+        }
+        _ => panic!("Only unit variants or named fields are supported in storage enums."),
+    };
+
+    let value_type = storage_attrs
+        .value_type
+        .as_ref()
+        .expect("value type required");
+
+    let getter_name = format_ident!("get_{}", variant_ident.to_string().to_snake_case());
+    let setter_name = format_ident!("set_{}", variant_ident.to_string().to_snake_case());
+
+    let storage_method = match storage_attrs.storage_type {
+        StorageType::Instance => quote! { instance },
+        StorageType::Persistent => quote! { persistent },
+        StorageType::Temporary => quote! { temporary },
+    };
+
+    let ttl_fn = match storage_attrs.storage_type {
+        StorageType::Persistent => quote! {
+            use stellar_axelar_std::ttl::extend_persistent_ttl;
+            extend_persistent_ttl(env, &key);
+        },
+        StorageType::Instance => quote! {
+            stellar_axelar_std::ttl::extend_instance_ttl(env);
+        },
+        _ => quote! {},
+    };
+
+    let key_construction = if field_names.is_empty() {
+        quote! { #enum_name::#variant_ident }
+    } else {
+        quote! { #enum_name::#variant_ident(#(#field_names),*) }
+    };
+
+    let param_list = if field_names.is_empty() {
+        quote! { env: &soroban_sdk::Env }
+    } else {
+        quote! { env: &soroban_sdk::Env, #(#field_names: #field_types),* }
+    };
+
+    quote! {
+        pub fn #getter_name(#param_list) -> Option<#value_type> {
+            let key = #key_construction;
+            let value = env.storage()
+                .#storage_method()
+                .get(&key);
```

Comment:
```suggestion
                .get::<_, #value_type>(&key);
```
auto-inference might not work for `Option<Option<...>>` case